plos_connector/main.py

```diff
@@ -0,0 +1,138 @@
+import requests
+import os
+import argparse
+
+
+cnvrg_workdir = os.environ.get("CNVRG_WORKDIR", "/cnvrg")
+
+class ValueMismatchError(Exception):
+    """Raise if number of urls and filenames do not match"""
+    def __init__(self):
+        super().__init__()
+
+    def __str__(self):
+        return "ValueMismatchError: The number of custom filenames must match the number of URLs."
+        
+
+class InvalidUrlError(Exception):
+    """Raise if URL is invalid"""
+    def __init__(self):
+        super().__init__()
+
+    def __str__(self):
+        return "InvalidUrlError: Failed to download from given URL, please check if provided URL leads to a downloadable PDF file."
+        
+
+
+
+def get_parameters():
+
+    parser = argparse.ArgumentParser(description="PLOS Connector")
+
+    parser.add_argument(
+        "--urls",
+        nargs="+",
+        action="store",
+        dest="urls",
+        required=True,
+        help="--- URLs to download journals from ---",
+    )
+
+    parser.add_argument(
+        "--filenames",
+        nargs="+",
+        action="store",
+        dest="filenames",
+        required=False,
+        default="j.pdf",
+        help="""--- Custom names of the pdf files ---""",
+    )
+    parser.add_argument(
+        "--local_dir",
+        action="store",
+        dest="local_dir",
+        required=False,
+        default=cnvrg_workdir,
+        help="""--- The path to save the dataset file to ---""",
+    )
+
+    parser.add_argument(
+        "--cnvrg_dataset",
+        action="store",
+        dest="cnvrg_dataset",
+        required=False,
+        default="None",
+        help="""--- the name of the cnvrg dataset to store in ---""",
+    )
+
+    return parser.parse_args()
+
+
+
+def download_journals(urls, filenames=None):
+
+    """Pulls journals as pdf files from a PLOS journal and saves them in the current directory
+        
+    
+    Args:
+            urls: string representing unique url(s) for journals to be downloaded
```

Comment:
Are both arguments lists of strings? Just wanted to confirm..

Reply:
Yes, both are list of strings